### ISSUE TITLE
Possible optimisations

### ISSUE THREAD

**Issue by wehlutyk** [ixxi-dante/nw2vec#20]
None of these are necessary, I'm just listing them to keep track of all the possible small-ish changes that might improve performance or reduce the need for resources:

Normalizations:

- [ ] [Batch normalization](http://arxiv.org/abs/1502.03167)
- [ ] [Self-normalization](http://arxiv.org/abs/1706.02515)

Connections:

- [ ] [Residual connections](http://arxiv.org/abs/1512.03385)

Reduction of necessary parameters and resources:

- [x] Mini-batching
- [ ] [Small sampling of neighbourhoods](http://arxiv.org/abs/1710.10568)

Noise for training:

- [ ] [Dropout](http://jmlr.org/papers/v15/srivastava14a.html)
- [ ] [Denoising](http://www.jmlr.org/papers/v11/vincent10a.html), [adapted to Variational Autoencoders](https://arxiv.org/abs/1511.06406)

Removing unneeded left-overs:

- [ ] [Bayesian compression](http://arxiv.org/abs/1711.06494)

**Comment by wehlutyk** [comment]
[How Powerful are Graph Neural Networks?](https://arxiv.org/abs/1810.00826) uses batch-normalisation at all layers.
